### PR TITLE
Clean up and add support for non-RGB bulbs

### PR DESCRIPTION
**Pull request by felipediel**
I am cleaning up the code and adding support for non-RGB bulbs.

I don't have the devices to test. Perhaps I missed something. Mind having a look and doing some tests before accepting the changes?

### CODE REVIEW

File: homeassistant/components/broadlink/light.py [hunk 1]
```diff
@@ -29,179 +30,105 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
...
-        if await self._async_send_packet({"pwr": 0}):
-            self._state = False
-            self.async_write_ha_state()
+        if self.supported_color_modes == {COLOR_MODE_BRIGHTNESS}:
+            state["bulb_colormode"] = BROADLINK_COLOR_MODE_WHITE
 
-    @abstractmethod
-    async def _async_send_packet(self, request):
-        """Send a packet to the device."""
+        elif ATTR_COLOR_MODE in kwargs:
```
Comment (L-I-Am) [inline homeassistant/components/broadlink/light.py:109]: color mode is never included in kwargs, that's why I implemented it originally to check if there's a temperature in the current kwargs or hs color in the current kwargs and to switch color mode depending on that (see lines 156 to 161 of original file).

I ran this locally and when setting a certain color, the only kwargs are: {'hs_color': (345.0, 100.0)}

File: homeassistant/components/broadlink/light.py [hunk 2]
```diff
@@ -29,179 +30,105 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
...
+            self._attr_supported_color_modes.add(COLOR_MODE_COLOR_TEMP)
+        if not self.supported_color_modes:
+            self._attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS}
+
+        self._update_state(data)
+
+    def _update_state(self, data):
+        """Update the state of the entity."""
+        if "pwr" in data:
+            self._is_on = data["pwr"]
```
Comment (L-I-Am) [inline homeassistant/components/broadlink/light.py]:
```suggestion
            self._attr_is_on = data["pwr"]
```

**Review by L-I-Am — CHANGES_REQUESTED** [review]
First of all: Thanks for the Merge request!

I ran it locally and got some errors at first, changing `self._is_on` to `self._attr_is_on` solved these errors. 

Setting brightness seems to be working, but setting a custom RGB color does nothing. Debugged it some more and found some problems in the color modes, addressed in code.

**Comment by felipediel** [comment]
Awesome, thank you! I am working on the changes.

**Comment by pr0fetul** [comment]
Hi,
As I am testing to see if the new code works, the setup is the following:
 - I am not home, I am testing it remotely. 
 - I have 5 LB1 s configured in the Broadlink App and added already to HA, but only 1 is active at this moment, depending on the rule of the first one that connects. It can be the 1st, 2nd, 5th and so on. I have never had all 5 at the same time working. 
 - I cannot remove and re-add them at this moment - next week when I am back home I can do this, but for now, I am relying on the entities that are already added. 

I have cloned all the files from the broadlink integration in my setup (broadlink folder in custom_components folder), cleared py_cache folder and restarted HA. 
Once back - these were the initial errors (logger is set on debug on the broadlink component):
```
2021-07-30 15:02:08 ERROR (SyncWorker_4) [homeassistant.loader] The custom integration 'broadlink' does not have a valid version key (None) in the manifest file and was blocked from loading. See https://developers.home-assistant.io/blog/2021/01/29/custom-integration-changes#versions for more details
```
```
2021-07-30 15:02:27 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Smart Bulb Set 2 for broadlink
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 293, in async_setup
result = await component.async_setup_entry(hass, self) # type: ignore
File "/usr/src/homeassistant/homeassistant/components/broadlink/__init__.py", line 35, in async_setup_entry
return await device.async_setup()
File "/usr/src/homeassistant/homeassistant/components/broadlink/device.py", line 110, in async_setup
update_manager = get_update_manager(self)
File "/usr/src/homeassistant/homeassistant/components/broadlink/updater.py", line 33, in get_update_manager
return update_managers[device.api.type](device)
KeyError: 'LB1'
```
I have added `"version": "1.0.0",` in the manifest.json file and restarted HA. Now the debug logs list these errors:
```
2021-07-30 15:16:02 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Smart Bulb Set 1' for broadlink integration not ready yet: [Errno -4000] Network timeout; Retrying in background
2021-07-30 15:16:02 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Smart Bulb Set 3' for broadlink integration not ready yet: [Errno -4000] Network timeout; Retrying in background
2021-07-30 15:16:02 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Smart Bulb 1' for broadlink integration not ready yet: [Errno -4000] Network timeout; Retrying in background
2021-07-30 15:16:04 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform opensky is taking over 10 seconds.
2021-07-30 15:16:07 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Smart Bulb Kitchen Balcony' for broadlink integration not ready yet: [Errno -4000] Network timeout; Retrying in background
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.switch] Error adding entities for domain switch with platform broadlink
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.switch] Error adding entities for domain switch with platform broadlink
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.remote] Error adding entities for domain remote with platform broadlink
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform broadlink
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.switch] Error while setting up broadlink platform for switch
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 258, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.switch] Error while setting up broadlink platform for switch
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 258, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.remote] Error while setting up broadlink platform for remote
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 258, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:07 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up broadlink platform for sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 258, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:14 ERROR (MainThread) [homeassistant.components.light] Error adding entities for domain light with platform broadlink
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
2021-07-30 15:16:14 ERROR (MainThread) [homeassistant.components.light] Error while setting up broadlink platform for light
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 258, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 458, in _async_add_entity
device_info = entity.device_info
File "/config/custom_components/broadlink/entity.py", line 55, in device_info
"connections": {(dr.CONNECTION_NETWORK_MAC, self._device.mac_address)},
AttributeError: 'BroadlinkDevice' object has no attribute 'mac_address'
```

**Comment by felipediel** [comment]
You cannot copy and paste the files at the custom components folder. This code is based on the latest dev branch, you need to set up a [dev environment](https://developers.home-assistant.io/docs/development_environment/). You can also git clone my repository and run script/setup.

**Review by L-I-Am — APPROVED** [review]
Tested locally and working for RGB-LB1 bulbs. Looks ok to me! thank you for putting in the time!

**Comment by felipediel** [comment]
Awesome, thank you! I think we can proceed without testing non-RGB, but it would be good to have. You just need to merge this PR and we continue there.

**Pull request merged**